pkg/resources/services/service.go

```diff
@@ -72,14 +77,18 @@ func NewServiceForCR(m *v1alpha1.Interconnect, requestCert bool) *corev1.Service
 			Namespace: m.Namespace,
 		},
 		Spec: corev1.ServiceSpec{
-			Type:     "LoadBalancer",
 			Selector: labels,
 			Ports:    servicePortsForRouter(m),
 		},
 	}
 	if requestCert {
 		service.Annotations = map[string]string{constants.CertRequestAnnotation: m.Name + "-cert"}
 	}
+	if m.Spec.DeploymentPlan.ServiceType == "ClusterIP" || m.Spec.DeploymentPlan.ServiceType == "NodePort" || m.Spec.DeploymentPlan.ServiceType == "LoadBalancer"  {
```

Comment:
lgtm, two questions:

1. When ServiceType is not provided, what is the default?
2. When ServiceType == "ClusterIP" should Spec.ClusterIP be set to "None"?

Reply:
1. ClusterIP is the default.
2. Spec.ClusterIP isn't set at all; when type is ClusterIP the controller will pick an IP and set that field. (Not sure if I am answering the right question?)